src/gui/creds/oauth.cpp

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright (C) by Olivier Goffart <ogoffart@woboq.com>
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+ * or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License
+ * for more details.
+ */
+
+#include <QDesktopServices>
+#include <QNetworkReply>
+#include <QTimer>
+#include "account.h"
+#include <json.h>
+#include "creds/oauth.h"
+#include "theme.h"
+
+
+namespace OCC
+{
+
+OAuth::~OAuth()
+{
+}
+
+static void httpReplyAndClose(QTcpSocket *socket, const char *code, const char *html)
+{
+    socket->write("HTTP/1.1 ");
+    socket->write(code);
+    socket->write("\r\nContent-Type: text/html\r\nConnection: close\r\nContent-Length: ");
+    socket->write(QByteArray::number(qstrlen(html)));
+    socket->write("\r\n\r\n");
+    socket->write(html);
+    socket->disconnectFromHost();
+}
+
+void OAuth::start()
+{
+    // Listen on the socket to get a port which will be used in the redirect_uri
+    if (!_server.listen(QHostAddress::LocalHost)) {
+        emit result(NotSupported, QString());
+        return;
+    }
+
+    if (!openBrowser())
+        return;
+
+    emit result(Waiting);
```

Comment:
This was usefull before OAuth in the browser based authentication,  but now it's not usefull anymore, i'll remove it.